pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -323,14 +324,39 @@ func (i *InfraProvider) DestroyBootstrap(dir string) error {
 
 	// TODO(padillon): start system if not running
 	if sys := clusterapi.System(); sys.State() == clusterapi.SystemStateRunning {
+		machineName := capiutils.GenerateBoostrapMachineName(metadata.InfraID)
+		machineNamespace := capiutils.Namespace
 		if err := sys.Client().Delete(context.TODO(), &clusterv1.Machine{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      capiutils.GenerateBoostrapMachineName(metadata.InfraID),
-				Namespace: capiutils.Namespace,
+				Name:      machineName,
+				Namespace: machineNamespace,
 			},
-		}); client.IgnoreNotFound(err) != nil {
+		}); err != nil {
 			return fmt.Errorf("failed to delete bootstrap machine: %w", err)
 		}
+
+		machineDeletionTimeout := 2 * time.Minute
+		logrus.Infof("Waiting up %v for machine deletion  %s/%s...", machineDeletionTimeout, machineNamespace, machineName)
+		machineContext, cancel := context.WithTimeout(context.TODO(), machineDeletionTimeout)
+		timer.StartTimer("Machine deletion")
```

Comment:
This timer seems like it is basically the same as the destroy bootstrap timer, as this is the only logic currently in destroy bootstrap. Maybe we don't need this?

Reply:
Indeed